x-pack/plugin/ccr/src/javaRestTest/java/org/elasticsearch/xpack/ccr/AbstractCCRRestTestCase.java

```diff
@@ -408,17 +387,6 @@ protected static void createAutoFollowPattern(
         assertOK(client.performRequest(request));
     }
 
-    /**
-     * Fix point in time when data stream backing index is first time queried.
-     * This is required to avoid failures when running test at midnight.
-     * (index is created for day0, but assertions are executed for day1 assuming different time based index name that does not exist)
-     */
-    private final LazyInitializable<Long, RuntimeException> time = new LazyInitializable<>(System::currentTimeMillis);
```

Comment:
The idea here is nice, but it doesn't completely avoid any issues, as the cluster(s) don't use this time supplier.